src/formatDate.js

```diff
@@ -8,7 +8,33 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromFormatCopy = [...fromFormat];
+  const toFormatCopy = [...toFormat];
+  const deleteLastElement = fromFormatCopy.pop();
+  const partOfDate = date.split(deleteLastElement);
+  const partsOfDate = {};
+
+  fromFormatCopy.forEach((part, index) => {
+    partsOfDate[part] = partOfDate[index];
+  });
+
+  if (toFormatCopy.includes('YY') && partsOfDate['YYYY']) {
+    partsOfDate['YY'] = partsOfDate['YYYY'].slice(-2);
+  }
+
+  if (toFormatCopy.includes('YYYY') && partsOfDate['YY']) {
+    partsOfDate['YYYY'] =
+      partsOfDate['YY'] < '30'
+        ? '20' + partsOfDate['YY']
+        : '19' + partsOfDate['YY'];
+  }
+
+  const newDeleteLastElement = toFormatCopy.pop();
+  const newDate = toFormatCopy
+    .map((part) => partsOfDate[part])
+    .join(newDeleteLastElement);
```

Comment:
Similar to the issue with `fromFormat`, using `newDeleteLastElement` as a delimiter for joining the date parts may not be correct. Ensure that the delimiter used matches the expected format in `toFormat`.